gtk4/src/subclass/editable.rs

```diff
@@ -0,0 +1,165 @@
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+use glib::GString;
+use gtk_sys;
+use libc::{c_char, c_int};
+
+use Editable;
+
+pub trait EditableImpl: ObjectImpl {
+    fn insert_text(&self, editable: &Self::Type, text: &str, length: i32, position: &mut i32);
+    fn delete_text(&self, editable: &Self::Type, start_position: i32, end_position: i32);
+    fn changed(&self, editable: &Self::Type);
+    fn get_text(&self, editable: &Self::Type) -> GString;
+    fn do_insert_text(&self, editable: &Self::Type, text: &str, length: i32, position: &mut i32);
+    fn do_delete_text(&self, editable: &Self::Type, start_position: i32, end_position: i32);
+    fn get_selection_bounds(
+        &self,
+        editable: &Self::Type,
+        start_position: &mut i32,
+        end_position: &mut i32,
```

Comment:
Should this maybe return an `Option<(i32, i32)>` instead of returning a bool and having two out parameters? That's at least what we would convert it to for the normal bindings code.

Reply:
> start_pos will be filled with the start of the selection and end_pos with end. If no text was selected both will be identical and FALSE will be returned.

This probably means that the most reasonable here would actually be a `Result<(i32, i32), i32>` on the Rust side. You should update the normal bindings accordingly too.

Reply:
Wouldn't an option makes more sense as editable doesn't really fail to get the selection bounds?

Reply:
It says that when `FALSE` is returned, both positions will be the same. So some position still has to be returned, it's just the same for both values

Reply:
right, but that from our side means we have "no selection bound" because both values are the same. If we are to provide a better API around this one, we should compare the values & return an Option. Does that sound sane to you? :)

Reply:
Wouldn't that lose the information about which same value both positions are? Does that same position have any actual meaning? I don't know this API or how it's used so you'll have to figure that one out :)

Reply:
Hmm very valid point, I will see if there are any usages of the API in the gtk-demos

Reply:
From the researches I have done it seems the api is used for two cases:
- Detect whether something is selected or not, in such case, a `NULL`  pointer is passed to both start_pos & end_pos and the usage only cares about whether it returns true or false. 

- Get the position of what's selected if there was anything selected: always checks first if the methods returns `true` and either adds/deletes text or emit a signal. 

So I think that same position value isn't actually useful as part of how this api is intended to be used. There's a `cursor-position` prop on the widgets that implements the Editable interface which should give that information to the user in case needed. I guess we should go with returning a Option<(i32, i32)> which looks the most accurate to me compared to the C API.

Reply:
That makes sense to me at least. Hard to say from the docs if that's the only intended usage, but worst case this can be improved later.

Reply:
Definitely. I will go ahead and implement that then :) Once I'm more or less happy with the progress on #56 I will start porting some of the custom & complex widgets from the C demo